### PR TITLE
📅 Add flag to suppress timestamps in logs

### LINKED ISSUE #428: User=nobody is discouraged by systemd as inherently unsafe
Please answer all the questions with enough information. All issues not following this template will be closed immediately.
If you are not sure if your question is truely a bug in V2Ray, please discuss it [here](https://github.com/v2fly/discussion/issues) first.

1) What version of V2Ray are you using (If you deploy different version on server and client, please explicitly point out)?

OS: Arch Linux
$ pacman -Q systemd v2ray
systemd 246.6-1
v2ray 4.32.1

2) What's your scenario of using V2Ray? E.g., Watching YouTube videos in Chrome via Socks/VMess proxy.

General testing.

3) What did you see? (Please describe in detail, such as timeout, fake TLS certificate etc)

Starting the systemd service throws a warning, stating that User=nobody is unsafe.

4) What's your expectation?

Running the systemd service as reasonably safe and secure as possible.

5) Please attach your configuration file (**Mask IP addresses before submit this issue**).

Server configuration:

```javascript

{
    "inbounds": [
        {
            "port": 10086, // Server Listening Port
            "protocol": "vmess",
            "settings": {
                "clients": [
                    {
                        "id": "b831381d-6324-4d53-ad4f-8cda48b30811"
                    }
                ]
            }
        }
    ],
    "outbounds": [
        {
            "protocol": "freedom"
        }
    ]
}

```

Client configuration:

```javascript

{
    "inbounds": [
        {
            "port": 1080, // SOCKS Proxy Port. Configure the browser proxy to use this port.
            "listen": "127.0.0.1",
            "protocol": "socks",
            "settings": {
                "udp": true
            }
        }
    ],
    "outbounds": [
        {
            "protocol": "vmess",
            "settings": {
                "vnext": [
                    {
                        "address": "server", // Server address: Change to your server hostname or ip address.
                        "port": 10086, // Server Listening Port
                        "users": [
                            {
                                "id": "b831381d-6324-4d53-ad4f-8cda48b30811"
                            }
                        ]
                    }
                ]
            }
        },
        {
            "protocol": "freedom",
            "tag": "direct"
        }
    ],
    "routing": {
        "domainStrategy": "IPOnDemand",
        "rules": [
            {
                "type": "field",
                "ip": [
                    "geoip:private"
                ],
                "outboundTag": "direct"
            }
        ]
    }
}

```

6) Please attach error logs, especially the bottom lines if the file is large. Error log file is usually at `/var/log/v2ray/error.log` on Linux.

Server error log:

```javascript

N/A
```

Client error log:

```javascript

N/A
```

7) Please attach access log. Access log is usually at '/var/log/v2ray/access.log' on Linux.

```javascript

N/A
```

8) Other configurations (such as Nginx) and logs.

9) If V2Ray doesn't run, please attach output from `--test`.

The command is usually `/usr/bin/v2ray/v2ray --test --config /etc/v2ray/config.json`, but may vary according to your scenario.

10) If V2Ray service doesn't run, please attach journal log.

Usual command is `journalctl -u v2ray`.

Please review your issue before submitting.

Systemd recently started to [discourage](https://github.com/systemd/systemd/commit/bed0b7dfc0070e920d00c89d9a4fd4db8d974cf0) the use of `User=nobody`. I have succesfully used a dynamic user configuration in the v2ray.service unit (see below), but that might not be the best way to deal with this. Lots of OS'es have an older systemd version and it's hard to determine/maintain when all the needed systemd options are supported. IMHO it would be easier to create a dedicated unprivileged user (e.g. v2ray) via systemd-sysusers.

```
[Unit]
Description=V2Ray Service
Documentation=https://www.v2fly.org/
After=network.target nss-lookup.target

[Service]
#User=nobody
DynamicUser=true
CacheDirectory=v2ray
LogsDirectory=v2ray
RuntimeDirectory=v2ray
StateDirectory=v2ray
CapabilityBoundingSet=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
AmbientCapabilities=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
NoNewPrivileges=true
ExecStart=/usr/bin/v2ray -config /etc/v2ray/config.json
Restart=on-failure
RestartPreventExitStatus=23

[Install]
WantedBy=multi-user.target
```

### PR DESCRIPTION
**Pull request by database64128**
This PR also updates the systemd unit files. Closes #428.

### CODE REVIEW

**Comment by xiaokangwang** [comment]
I would like to advise you not to use global flags. V2Ray must not assume the user is using main/main.go as the entry point. 

Multi-instance support for V2Ray is planned in V5. This means all settings are recommended to be local to instance, and any new global state is strictly not recommended.

**Comment by database64128** [comment]
@xiaokangwang I don't want to use global flags either. But there's no other way. Also I'm not assuming using `main/main.go` as the entry point. The logger behaves just the same without the main package.

**Comment by database64128** [comment]
The code changes in this PR have been relicensed under AGPLv3 and applied to a downstream fork [Shadowsocks-NET/v2ray-go](https://github.com/Shadowsocks-NET/v2ray-go). Due to license requirements, these changes may not be applied to v2fly/v2ray-core.

File: release/config/systemd/system/v2ray.service [hunk 1]
```diff
@@ -4,11 +4,12 @@ Documentation=https://www.v2fly.org/
 After=network.target nss-lookup.target
 
 [Service]
-User=nobody
```
Comment (hldh214) [inline release/config/systemd/system/v2ray.service:7]: 我觉得这里应该用 `DynamicUser=yes`

ref: 
https://github.com/systemd/systemd/commit/bed0b7dfc0070e920d00c89d9a4fd4db8d974cf0
https://github.com/systemd-cron/systemd-cron/commit/be4aea647bce9a0f26c9b8126f7883dd5c0b60a2

**Pull request closed without merging**